File: MATTHEW/leetcode/AIcoding/4.GPT-4o/erlang_solutions/1833-maximum-ice-cream-bars.erl
-module(solution).
-export([maxIceCream/2]).

maxIceCream(Money, Costs) ->
    SortedCosts = lists:sort(Costs),
    max_ice_cream_helper(Money, SortedCosts, 0).

max_ice_cream_helper(_, [], Count) ->
    Count;
max_ice_cream_helper(Money, [Cost | Rest], Count) when Money >= Cost ->
    max_ice_cream_helper(Money - Cost, Rest, Count + 1);
max_ice_cream_helper(_, _, Count) ->
    Count.

File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/1833-maximum-ice-cream-bars.rs
impl Solution {
    pub fn max_ice_cream(costs: Vec<i32>, coins: i32) -> i32 {
        let mut costs = costs;
        costs.sort_unstable();
        let mut remaining = coins;
        let mut count = 0;
        for cost in costs {
            if remaining >= cost {
                remaining -= cost;
                count += 1;
            } else {
                break;
            }
        }
        count
    }
}